app/models/order_detail.rb

```diff
@@ -288,9 +287,11 @@ def can_be_viewed_by?(user)
   scope :for_facilities, ->(facilities) { joins(:order).where("orders.facility_id in (?)", facilities) unless facilities.nil? || facilities.empty? }
   scope :for_products, ->(products) { where("order_details.product_id in (?)", products) unless products.blank? }
   scope :for_owners, lambda { |owners|
-    joins(:account)
-      .joins("INNER JOIN account_users on account_users.account_id = accounts.id and user_role = 'Owner'")
-      .where("account_users.user_id in (?)", owners) unless owners.blank?
+    unless owners.blank?
+      joins(:account)
+        .joins("INNER JOIN account_users on account_users.account_id = accounts.id and user_role = 'Owner'")
```

Comment:
I like pulling the conditional up. Now that's it's at a block level, would it be clearer to use `if owners.any?`/`present?`?

Also, I feel like this could be cleaned up using something like `merge(AccountUser.owners)`, but I haven't taken the time to figure out exactly what that would be.